wsd/COOLWSD.cpp

```diff
@@ -4251,7 +4259,14 @@ void COOLWSD::processFetchUpdate()
         FetchHttpSession->setFinishedHandler([](const std::shared_ptr<http::Session>& httpSession) {
             std::shared_ptr<http::Response> httpResponse = httpSession->response();
 
-            FetchHttpSession.reset();
+            // Retire.
```

Comment:
Wow - really unhappy about this now I look. Why is this processFetchUpdate happening  in the 'mainWait' SocketPoll - and then using the WebServerPoll() to do its work ? that seems nonsensical.

Surely we should do the work in the mainWait poll in COOLWSD.cpp to avoid cross-thread horrors (and fouling up the webseverpoll with that ;)

Reply:
Oh, I didn't realize we were in `mainWait`. Sorry, I'm not the original author of this feature.

I'm trying to avoid resetting what seems like the main reference to the Session from which the callback is done.
My concern is that if this is the last reference, we can segfault when we return from the callback and the local reference gets released _before_ the last member access. Something that the compiler controls and we can't force it to see this hard dependency.

Will fix the mismatch of polls. Thanks for catching it!

Reply:
In these cases - we should always hold references to 'this' all the way down the stack; and ideally in the SocketPoll / Socket infrastructure above too.

Reply:
Agreed. This is why I added the assertions on the `use_count()` of the `shared_ptr`, so we can catch when there is a violation. Not sure if there is a better way to enforce what you suggest otherwise.